### ISSUE TITLE
making groff to compile Hebrew and and Arabic bidi languages

### ISSUE THREAD

**Issue by OrionRandD** [user18130814200115-2/groffscripts#5]
Hi.
Do you know how can I compile a .ms groff macro that has got Hebrew and Arabic languages into .pdf or .ps?
I have see that distroff wich is now supported by groff can do this. But I have no idea wich is the groff command to compile the the .ms to pdf showing Hebrew and Arabic in the pdf of ps...

**Comment by user18130814200115-2** [comment]
Okay, so compiling a document like this involves a couple of steps:
1. You need to get a .ttf font file which contains the glyps you are going to use.
2. You need to make this font accessible to groff by following [these steps](https://www.schaffter.ca/mom/momdoc/appendices.html)
3. You need to preprocess the document with `preconv -eutf8`, this will make the file readbale for groff, I recommend running processord like `refer` **before** you use preconv.
 4. With the encoded text from preconv, the process to get groff to work is the same as always, this sould get you the following command:

`cat FILENAME | preconv -euft8 | groff -Tpdf > FILENAME.pdf`


You should know a couple of things though.
Since groff does not naively support non-ascii characters, preprocessors may act funky, and you likely cannot write macros in arabic/hebrew. I wrote a blogpost about these issues a while ago which you can read [here](https://blog.wester.digital/latex-groff-and-neatroff.html)

Lastly, there is getting groff to print text from right to left which is a big pain.
In short, groff does not support this and outside of [some hacks](https://lists.gnu.org/archive/html/groff/2006-12/msg00018.html) you cannot properly get it to work.
I recommend checking out [neatroff](https://github.com/aligrudi/neatroff) as it was designed with non-english languages in mind and therefore supports RTL printing and UTF8 natively.
Again, see [the blog post](https://blog.wester.digital/latex-groff-and-neatroff.html) for more info on neatroff.

**Comment by user18130814200115-2** [comment]
I am closing this issue, as it is not relevant to this particular repo, but feel free to contact me again if you have more questions about getting your compilation to work.

**Comment by OrionRandD** [comment]
Thx for the hints :)
